### PR TITLE
Add Owasp SameSite cookie params

### PR DESCRIPTION
**Pull request by DBaker85**
Many sites now require the Samesite cookie Param for secure requests.

[https://www.owasp.org/index.php/SameSite](https://www.owasp.org/index.php/SameSite)

This PR 
- adds the Owasp required `sameSite` cookie parameter.
- fixes paths to work cross environment by splitting each segment of path.
- fixes e2e with a newer version of ts-node

### CODE REVIEW

**Comment by CunningFatalist** [comment]
Hello DBaker85,

thanks a bunch for your pull request. This looks like a great feature, and we really appreciate it. There are some things we would like to change about this, though.

First, I guess your IDE auto formatted all of our files. I know that our style guide is old and not too beloved, but we use [idiomatic.js](https://github.com/rwaldron/idiomatic.js/) and would like to stick with it, because we don't want to obscure our GIT history.

Then, [according to MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie), `SameSite` only accepts two values: `'Strict'` and `'Lax'` (capitalized). Therefore, we should write `sameSite?: 'Strict' | 'Lax'` both in the documentation and in the library instead of `sameSite: string`.

We talked about writing tests for this, but this is hardly possible with our current test setup. We also talked about the technology being experimental, but everything that improves security is great in our book.

Also, feel free to add yourself to the list of contributors in both readme files and the `package.json` :smile:

Cheers

**Comment by DBaker85** [comment]
Yeah sorry about the auto formatting guys. I fixed it now :)

File: lib/cookie-service/cookie.service.ts [hunk 1]
```diff
@@ -81,20 +81,22 @@ export class CookieService {
...
-   * @param path    Cookie path
-   * @param domain  Cookie domain
-   * @param secure  Secure flag
+   * @param name     Cookie name
+   * @param value    Cookie value
+   * @param expires  Number of days until the cookies expires or an actual `Date`
+   * @param path     Cookie path
+   * @param domain   Cookie domain
+   * @param secure   Secure flag
+   * @param sameSite OWASP samesite token `lax` or `strict`
```
Comment (CunningFatalist) [inline lib/cookie-service/cookie.service.ts]: This should also be capitalized: 

```
   * @param sameSite OWASP samesite token `Lax` or `Strict`
```

File: lib/cookie-service/cookie.service.ts [hunk 2]
```diff
@@ -124,6 +126,10 @@ export class CookieService {
       cookieString += 'secure;';
     }
 
+    if (sameSite) {
```
Comment (CunningFatalist) [inline lib/cookie-service/cookie.service.ts]: I know the style guide is a little odd 😄 but this should be:

```
  if ( sameSite ) {
```

File: package.json [hunk 3]
```diff
@@ -100,7 +103,7 @@
     "protractor": "~5.1.0",
     "rimraf": "^2.6.1",
     "rxjs": "^5.1.0",
-    "ts-node": "~2.0.0",
+    "ts-node": "~3.3.0",
```
Comment (CunningFatalist) [inline package.json]: I see no reason to bump up the `ts-node` version, everything will still work with `~2.0.0`.

File: scripts/compile.js [hunk 4]
```diff
@@ -23,7 +23,7 @@ async.waterfall([
 });
```
Comment (CunningFatalist) [inline scripts/compile.js]: I also see no reason to change the compile script in this PR, just revert the changes 😉

Comment (DBaker85) [reply scripts/compile.js]: I changed this one because it does not work under windows. Changing all the instances of `path` allows this to work in windows, but I can revert it if you prefer

Comment (CunningFatalist) [reply scripts/compile.js]: Yes, please revert it for now. I didn't know that it doesn't work with Windows, we should probably fix that. However, this would have to be a different pull request and I would suggest using `path.resolve` in that case (see [Path @ NodeJS docs](https://nodejs.org/docs/latest/api/path.html)).

File: package.json (hunk 3 above)
Comment (CunningFatalist) [reply package.json]: See my other comment. This would ideally be part of another Pull Request that fixes a new issue (something like "cannot compile library in Windows") :wink:

File: lib/cookie-service/cookie.service.ts (hunk 1 above)
Comment (DBaker85) [reply lib/cookie-service/cookie.service.ts]: Done :)

File: lib/cookie-service/cookie.service.ts (hunk 2 above)
Comment (DBaker85) [reply lib/cookie-service/cookie.service.ts]: Done :)

File: package.json (hunk 3 above)
Comment (DBaker85) [reply package.json]: I agree. Reverted :)

File: scripts/compile.js (hunk 4 above)
Comment (DBaker85) [reply scripts/compile.js]: I agree. Reverted :)
